dask/dataframe/core.py

```diff
@@ -1461,17 +1461,76 @@ def var(self, axis=None, skipna=True, ddof=1, split_every=False, dtype=None, out
                                     axis=axis, skipna=skipna, ddof=ddof)
             return handle_out(out, result)
         else:
-            num = self._get_numeric_data()
-            x = 1.0 * num.sum(skipna=skipna, split_every=split_every)
-            x2 = 1.0 * (num ** 2).sum(skipna=skipna, split_every=split_every)
-            n = num.count(split_every=split_every)
-            name = self._token_prefix + 'var'
-            result = map_partitions(methods.var_aggregate, x2, x, n,
-                                    token=name, meta=meta, ddof=ddof)
+            if self.ndim == 1:
+                result = self._var_1d(self, skipna, ddof, split_every)
+            elif len(self._meta_nonempty.select_dtypes(include=[np.timedelta64]).columns) > 0:
+                result = self._var_mixed(skipna, ddof, split_every)
+            else:
+                result = self._var_numeric(skipna, ddof, split_every)
+
             if isinstance(self, DataFrame):
                 result.divisions = (min(self.columns), max(self.columns))
             return handle_out(out, result)
 
+    def _var_numeric(self, skipna=True, ddof=1, split_every=False):
+        num = self.select_dtypes(include=['number', 'bool'])
+
+        values_dtype = num.values.dtype
+        array_values = num.values
+
+        if not np.issubdtype(values_dtype, np.number):
+            array_values = num.values.astype('f8')
+
+        var = da.nanvar if skipna or skipna is None else da.var
+        array_var = var(array_values, axis=0, ddof=ddof, split_every=split_every)
+
+        name = self._token_prefix + 'var-numeric' + tokenize(num, split_every)
+        cols = num._meta.columns if is_dataframe_like(num) else None
+
+        var_shape = num._meta_nonempty.values.var(axis=0).shape
+        array_var_name = (array_var._name,) + (0,) * len(var_shape)
+
+        layer = {(name, 0): (methods.wrap_var_reduction, array_var_name, cols)}
+        graph = HighLevelGraph.from_collections(name, layer, dependencies=[array_var])
+
+        return new_dd_object(graph, name, num._meta_nonempty.var(), divisions=[None, None])
+
+    def _var_mixed(self, skipna=True, ddof=1, split_every=False):
+        data = self.select_dtypes(include=['number', np.timedelta64, 'bool'])
+
+        vars = [self._var_1d(data[col_idx], skipna, ddof, split_every) for col_idx in data._meta.columns]
+        var_names = [(v._name, 0) for v in vars]
+
+        name = self._token_prefix + 'var-mixed-' + tokenize(data, split_every)
+
+        layer = {(name, 0): (methods.wrap_var_reduction, var_names, data._meta.columns)}
+        graph = HighLevelGraph.from_collections(name, layer, dependencies=vars)
+
+        return new_dd_object(graph, name, data._meta_nonempty.var(), divisions=[None, None])
+
+    def _var_1d(self, column, skipna=True, ddof=1, split_every=False):
+        is_timedelta = is_timedelta64_dtype(column._meta)
+
+        if is_timedelta:
+            column = column.dropna().astype('i8')
```

Comment:
Is this working around https://github.com/pandas-dev/pandas/issues/18880? Ideally we'd like to follow skipna, but that bug may prevent it.

Reply:
@TomAugspurger looks like. I've looked into your PR for this bug and this what ideally should be done here as well:
```python
    if is_timedelta64_dtype(values) and not skipna:
        values = values.astype('float64')
        values[mask] = np.nan
```
But I suspect it's going to slow down things. What do you think?

Reply:
I just realized, we aren't using pandas' `nanstd`, so we don't need to worry about pandas-dev/pandas#18880 hopefully.

I'm not too concerned about the cost of computing the mask. Pandas does that internally as well, so it won't be outrageously slower (and this branch is only for `not skipna`, so it's not the default). Let's do something like that.

Reply:
Done.